src/main/resources/templates/fragments/exclude.html

```diff
@@ -62,10 +68,11 @@ <h1 class="font-weight-bold text-dark pt-2 mb-0 d-inline-block">Exclude ({{group
                 <tr ng-repeat="e in pagedItemsExclude[currentPageExclude]" class="table table-sm">
                     <td class="overflow-auto py-1 px-3">
                         <input class="position-relative ml-0 blank-checkbox" type="checkbox"
-                               aria-label="Remove {{i.name}} from the exclude."
+                               aria-label="Remove {{e.name}} from the exclude."
                                ng-keydown="clickCheckbox($event, e)"
                                ng-model="membersInCheckboxList[e.uhUuid]"
-                               ng-click="$event.stopPropagation()">
+                               id="{{'exclude-checkbox-' + e.uhUuid}}"
+                               ng-click="$event.stopPropagation();toggleCheckAllSelection('Exclude', 'single')">
```

Comment:
I find it a bit weird to see toggleCheckAllSelection being called for one of the singular checkboxes here. You might be able to simplify everything by keeping constant track of the number of checkboxes that are currently checked as a scope variable. 

In short, try to simplify the logic by using a scope variable to keep track of the total number of checkboxes currently selected. Next, use a second function maybe called updateTotalSelectedCheckboxes() that is appended to the ng-click so that every checkbox is able to change the total count. Overall, I want to be able to avoid using $scope.watch because it can get very laggy.

Reply:
I think I did attempt a implementation of keeping track of the total selected, but ran into a problem with total being triggered incorrectly. I can try again.